cpu/microarchitectures.json

```
@@ -1375,6 +1375,35 @@
         ]
       }
     },
+    "graviton": {
+      "from": ["aarch64"],
+      "vendor": "ARM",
+      "features": [
+          "fp",
+          "asimd",
+          "evtstrm",
+          "aes",
+          "pmull",
+          "sha1",
+          "sha2",
+          "crc32",
+          "cpuid"
+      ],
+      "compilers" : {
+          "gcc": [
+              {
+                  "versions": ":",
+                  "flags" : "-march==armv8-a+crc+crypto -mtune=cortex-a72"
+              }
+          ],
+          "clang" : [
+              {
+                  "versions": ":",
+                  "flags" : "-march=armv8-a+crc+crypto"
+              }
+          ]
+      }
+    },
     "graviton2": {
       "from": ["aarch64"],
```

@TomTheBear You should change this to to correctly capture the relation between Graviton and Graviton2:

```suggestion
      "from": ["graviton"],
```

---

I understand your request, but Graviton and Graviton 2 are completely different designs and both base on `aarch64`. Graviton 2 is not an enhanced version of Graviton. To be clear, Graviton is an [ARM Cortex-A72](https://www.extremetech.com/computing/281338-amazon-launches-new-arm-based-custom-graviton-cpu) core and Graviton 2 is an [ARM Neoverse N1](https://www.anandtech.com/show/15189/amazon-announces-graviton2-soc-along-with-new-aws-instances-64core-arm-with-large-performance-uplifts) design. It's a general question whether these designs should be added to the JSON document and the other ARM entries should have them as parent.

It is not consitent in the current JSON and there is no real guideline here how to add entries. While the track can be followed for Intel architectures (with some strange settings like for `mic_knl` which should be either `x86_64` or better missing [`silvermont`](https://www.realworldtech.com/knights-landing-details/), it's not the case for AMD (`k10` -> `bulldozer` or `excavator` -> `zen` where `bulldozer` and `zen`base on `x86_64`).

---

I think we'll need some input from @alalazo here...

---

I think it’s fine to have these as separate chips deriving from `aarch64`, as @TomTheBear originally did, since they’re not compatible.  I was thinking about this yesterday and I looked at the flags — you can see graviton is not a subset of graviton2 from the flags. 

Generally what we want with the `from` relationship is a feature subset relationship, which means the microarchitectures will be compatible. We don’t implement it as a subset check, though, because new features (at least in the Intel world) can subsume old ones. So a chip with `ssse3` has the `sse3` feature, it just doesn’t say so.  `from` tells archspec it’s still compatible, and we have some aliasing support in the JSON so you can query for `sse3` and get `True` even on uarches with `ssse3`.

Anyway, I think this is fine as is for now, as the intent is to use the *labels* (`graviton`, `graviton2`, etc.) to label binaries. We can improve what archspec knows about them over time.

It did occur to me that maybe we should have intermediate nodes for arm, for things like `armv8-a`, `armv8.2-a`, etc. those would derive from `aarch64`, and the specific chips would derive from them. Then we’d have nicer names for more generic targets. It seems like it’s still worth having the specific chip entries since they add features to the uarch, but I could imagine people wanting to build something slightly more generic but still optimized for the particular arm spec version.

@TomTheBear does that sound more reasonable to you?

@alalazo? @becker33?

---

It wasn't clear to me that `graviton2` isn't a successor to `graviton` (as the name suggests, I guess), so what @tgamblin proposes makes sense to me.

With that in mind, I think this PR is good to go, except maybe the remarks about the compiler flags?